### PR TITLE
Ability to upload contents of directory only without creating the parent directory in Cloudinary.

### PR DESCRIPTION
**Pull request by chrissnyder2337**
### Brief Summary of Changes

Add the ability to upload the contents of a local directory without requiring that the parent directory (the `local_folder`) be created in Cloudinary. 

Without this change, it is impossible to use `cld upload_dir` to upload a local directory's contents to Cloudinary without also creating a directory/folder within Cloudinary that also has the same name.

This was implemented by introducing new `--exclude-dir-name (-e)` command option that allowa the user to decide if the local parent directory should be created in Cloudinary, or just its contents.

**Example without flag**
Upload the local folder, `my_images`, and all its contents and sub-folders to the Cloudinary folder `my_images_on_cloudinary`, overwriting existing files. This example was pulled from the [Cloudinary CLI documentation](https://cloudinary.com/documentation/cloudinary_cli#upload_dir).  

```console
cld upload_dir -f my_images_on_cloudinary -o overwrite true my_images
```

This would result in the following folder structure in Cloudinary:

```
    Home
    +--my_images_on_cloudinary
    |  +-- my_images
    |  |   +-- <all assets and sub-folders of my_images>
```

**Example using new -e (--exclude-dir-name) flag**
Upload the *contents* of local folder, `my_images`, and all sub-folders to the Cloudinary folder `my_images_on_cloudinary`, overwriting existing files. 

```console
cld upload_dir -f my_images_on_cloudinary -e -o overwrite true my_images
```
This would result in the following folder structure in Cloudinary:

```
    Home
    +-- my_images_on_cloudinary
    |  +-- <all assets and sub-folders of my_images>
```

#### What does this PR address?
- [x] GitHub issue (#64)
- [ ] Refactoring
- [x] New feature
- [ ] Bug fix
- [x] Adds more tests

#### Are tests included?
- [x] Yes
- [ ] No

#### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I ran the full test suite before pushing the changes and all the tests pass.

### CODE REVIEW

**Comment by const-cloudinary** [comment]
@chrissnyder2337 Thank you for the PR! It is really a good use case.
We had an internal discussion about the parameter name and we got to the following:
It would be great if you could change the parameter name to: 
`--use-parent-dir` and set it to True by default (current behavior)
Add a few tests under:
https://github.com/cloudinary/cloudinary-cli/blob/master/test/test_modules/test_cli_upload_dir.py

**Comment by chrissnyder2337** [comment]
@const-cloudinary I have changed the parameter's name to `--use-parent-dir` and set it to default to true. 

Due to the way that boolean flags work, and the desire to keep all existing functionality when the new command line option was not provided, I also needed to add a way to turn it off by adding a `--dont-use-parent-dir(-P)` option. 

See https://click.palletsprojects.com/en/8.1.x/options/#boolean-flags

**Comment by chrissnyder2337** [comment]
@const-cloudinary I have updated this PR's description to reflect the change of parameters.

**Comment by chrissnyder2337** [comment]
@const-cloudinary I have just pushed an additional change to this PR to add a couple of tests and tweaked the implementation to account for a bug I found.

**Comment by chrissnyder2337** [comment]
FYI: I have updated this PR's description to note the added tests.

**Comment by const-cloudinary** [comment]
@chrissnyder2337 , thanks for the fixes and tests.
Let me just check with our PM team whether they are OK with the name of the parameter.

**Comment by const-cloudinary** [comment]
@chrissnyder2337 , after another round of discussions, having a default parameter with a true value, which practically has no sense to keep, we came to this name:
`-e`, `--exclude-dir-name`, which removes the directory name.
Hope it's not too much work for you to change it.
Sorry for confusion.

**Comment by chrissnyder2337** [comment]
@const-cloudinary I get it, naming things is hard.

I have changed the option to `-e`/`--exclude-dir-name` and updated the tests accordingly

**Review by const-cloudinary — APPROVED** [review]
Looks great!

Will wait for approval from our docs team.

File: cloudinary_cli/modules/upload_dir.py
Comment (jackieros) [inline cloudinary_cli/modules/upload_dir.py:29]:
```suggestion
        help="Don't include the selected parent directory name in the public ID path of the uploaded files."
```

File: cloudinary_cli/modules/upload_dir.py
Comment (jackieros) [inline cloudinary_cli/modules/upload_dir.py:30]:
```suggestion
             "This ensures that the public ID paths of the uploaded assets will be directly under the specified --(f)older, avoiding an extraneous level in the path."
```

File: cloudinary_cli/modules/upload_dir.py
Comment (jackieros) [inline cloudinary_cli/modules/upload_dir.py:31]:
```suggestion
             "When this option is used, the contents of the parent directory are uploaded instead of the parent directory itself and thus the name of the specified parent directory is not included in the pubic ID path of the uploaded assets.)
```

Comment (jackieros) [reply cloudinary_cli/modules/upload_dir.py:31]: @const-cloudinary - I'd like to suggest that we remove the description lines 29 & 30, and keep only this slightly expanded version of line 31.  I think the clarification that the option uploads the contents instead of the directory is the most clear explanation and with the additional phrase about it "thus" not including the parent directory name, I think we explain everything necessary in phrasing that works well for both fixed folder mode & dynamic folder mode.  

(Lines 29 & 30 get messy/confusing when trying to make the wording fit both modes, and don't add new info.  We'll include examples in the docs to provide additional clarity).

File: cloudinary_cli/modules/upload_dir.py
Comment (jackieros) [inline cloudinary_cli/modules/upload_dir.py:25]: @const-cloudinary - Suggest changing this help text to: 

The path where you want to upload the assets.  The path you specify will be pre-pended to the public IDs of the uploaded assets.  You can specify a whole path, for example path1/path2/path3.

File: cloudinary_cli/modules/upload_dir.py
Comment (jackieros) [inline cloudinary_cli/modules/upload_dir.py:26]:
```suggestion
             "If your product environment uses fixed folder mode, then any folders that do not exist are automatically created.")
```

**Review by jackieros — COMMENTED** [review]
@const-cloudinary Reviewed help texts.

**Comment by chrissnyder2337** [comment]
@const-cloudinary @jackieros Let me know which, if not all, of the documentation suggestions to include in this PR.

File: cloudinary_cli/modules/upload_dir.py
Comment (const-cloudinary) [inline cloudinary_cli/modules/upload_dir.py:31]:
```suggestion
             "When this option is used, the contents of the parent directory are uploaded instead of the parent directory itself and thus the name of the specified parent directory is not included in the pubic ID path of the uploaded assets.")
```

File: test/test_modules/test_cli_upload_dir.py
Comment (const-cloudinary) [inline test/test_modules/test_cli_upload_dir.py:46]:
```suggestion
        self.assertIn("as " + self.CLD_UPLOAD_DIR + "/test_sync/", result.output)
```

File: test/test_modules/test_cli_upload_dir.py
Comment (const-cloudinary) [inline test/test_modules/test_cli_upload_dir.py:54]:
```suggestion
        self.assertIn("as " + self.CLD_UPLOAD_DIR, result.output)
        self.assertNotIn("as " + self.CLD_UPLOAD_DIR + "/test_sync/", result.output)
```

**Comment by const-cloudinary** [comment]
Fixes #64

**Pull request merged**
